lib/perso.ml

```diff
@@ -3527,11 +4188,32 @@ let eval_transl conf base env upp s c =
       if upp then Utf8.capitalize_fst r else r
   | _ -> Templ.eval_transl conf upp s c
 
+let level_in_list level lev_list =
+  let level = if List.hd lev_list >= 0 then level else -level in
+  if List.mem level lev_list then Some level else None
+
+let level_less_or_eq_list level lev_list =
+  if List.hd lev_list >= 0 then
+    let rec loop lev_list =
+      match lev_list with
+      | [] -> None
+      | lev :: lev_list -> if lev <= level then Some lev else loop lev_list
+    in
+    loop lev_list
+  else
+    let rec loop lev_list =
+      match lev_list with
+      | [] -> None
+      | lev :: lev_list -> if lev >= -level then Some lev else loop lev_list
+    in
```

Comment:
you can factorize this `if ... else ...` if you take the absolute value of `lev`:
```ocaml
| lev :: lev_list -> if abs lev <= level then Some lev else loop lev_list
```
Also the loop can be replaced by something like:
```ocaml
List.find_opt (fun lvl -> abs lvl <= level) lev_list
```